maven_iuvs/echelle.py

```diff
@@ -1931,26 +2136,29 @@ def prior_transform(u, param_bounds):
                 
             """
             x = np.array(u)
+            for i in range(len(x)):
+                x[i] = uniform(u[i], param_bounds[i][0], param_bounds[i][1])
             
-            x[0] = uniform(u[0], param_bounds[0][0], param_bounds[0][1]) # DN for H line: Probably integrates between 1 and 10000 DN? 400 is a "good guess"
-            x[1] = uniform(u[1], param_bounds[1][0], param_bounds[1][1]) # DN for D line: set to the same.
-            x[2] = uniform(u[2], param_bounds[2][0], param_bounds[2][1]) # Line center for ly alpha of H
-            x[3] = uniform(u[3], param_bounds[3][0], param_bounds[3][1]) # bg_m_guess
-            x[4] = uniform(u[4], param_bounds[4][0], param_bounds[4][1]) # bg_b_guess; usually the median of the spectrum, could be anywhere from 0 (no background) to maybe 3000 (higher than emissions)
             return x
 
         # List of arguments for loglikelihood
         loglike_args = [wavelengths, edges, CLSF, spec, unc, 1, BU_bg]
 
         # List of arguments for prior_transform
-        ptf_args = [ [[param_initial_guess[0]/1000, param_initial_guess[0]*1000], # Total DN, H
-                      [param_initial_guess[1]/1000, param_initial_guess[1]*1000],  # Total DN, D
-                      [param_initial_guess[2]-0.5, param_initial_guess[2]+0.5], # H lyman alpha center in nm
-                      [param_initial_guess[3]-1e4, param_initial_guess[3]+1e4], # background slope
-                      [param_initial_guess[4]-1e4, param_initial_guess[4]+1e4] # Background offset
+        ptf_args = [ [[pig[0]/10, pig[0]*10], # Total DN, H
+                      [-pig[1]/10, pig[1]*10],  # DN, D
+                      [-pig[2]/2, pig[2]*2], # DN, IPH
+                      [pig[3]-0.02, pig[3]+0.02], # H lyman alpha center in nm: 121.547 -- 121.587
+                      [pig[4]-0.03, pig[4]+0.03], # IPH ly a center in nm
+                      [pig[5]-0.003, pig[5]+0.003], # IPH width
+                      [pig[6]-1e4, pig[6]+1e4], # background slope
+                      [pig[7]-1e4, pig[7]+1e4] # Background offset
                      ] 
                    ]
 
+        print(f"Initial guess for the IPH will be: area {ptf_args[0][2]}, " 
+                + f"center {ptf_args[0][4]}, width {ptf_args[0][5]}")
+
         if approach=="dynamic":
             dsampler = d.DynamicNestedSampler(loglikelihood, prior_transform, len(ptf_args[0]), 
```

Comment:
does the sign argument to loglikelihood need to change for dynesty?

Reply:
Fixed in 4629870a59d41724507a2ff402c35f9a9f594b41

Reply:
I think if the default sign has been changed to minus, the function should be called negloglikelihood

Reply:
There's no default, it uses -1 for dynesty and +1 for scipy. Since there is no obvious negative value in it as written I think calling it negloglikelihood would be confusing

Reply:
something like keeping the current implementation of loglikelihood() and implementing negloglikelihood() as

> `def negloglikelihood(*args, **kwargs):`
> `     return -loglikelihood(*args, **kwargs)`

Reply:
Ah, this is different from the previous message and I like this solution better.